crates/ty_python_semantic/resources/mdtest/narrow/type.md

```diff
@@ -54,7 +54,7 @@ def _(x: A | B):
         reveal_type(x)  # revealed: A | B
 ```
 
-## No narrowing for custom `type` callable
+## No special narrowing for custom `type` callable
```

Comment:
Maybe let's have the custom `type()` function imported from a stub, so we can keep the clear original intent of the test here, without RTI making it more confusing.